src/dbnode/server/server.go

```diff
@@ -483,6 +494,7 @@ func Run(runOpts RunOptions) {
 
 	// Apply pooling options.
 	opts = withEncodingAndPoolingOptions(cfg, logger, opts, cfg.PoolingPolicy)
+	opts = opts.SetIndexOptions(opts.IndexOptions().SetMmapReporter(mmapReporter))
```

Comment:
I'm going to move index options construction earlier to more naturally set the mmap reporter on it when it's first created.